mjpeg-view/src/main/java/com/github/niqdev/mjpeg/Mjpeg.java

```diff
@@ -94,15 +97,25 @@ public Mjpeg sendConnectionCloseHeader() {
     }
 
     @NonNull
-    private Observable<MjpegInputStream> connect(String url) {
+    private Observable<MjpegInputStream> connect(String url, String username, String password) {
         return Observable.defer(() -> {
             try {
                 HttpURLConnection urlConnection = (HttpURLConnection) new URL(url).openConnection();
                 urlConnection.setRequestProperty("Cache-Control", "no-cache");
+
                 if (sendConnectionCloseHeader) {
                     urlConnection.setRequestProperty("Connection", "close");
                 }
 
+                if (urlConnection.getResponseCode() == HttpURLConnection.HTTP_UNAUTHORIZED && !TextUtils.isEmpty(username) && !TextUtils.isEmpty(password)) {
```

Comment:
if is worth it, maybe extract it in a method and add comment for why is needed, otherwise is fine.
The important thing is that it must be retrocompatible

Reply:
Agreed, I will do it later. Thanks for your feedback